CMakeLists.txt

```diff
@@ -123,7 +123,10 @@ if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
     set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
```

Comment:
Should an additional `if (NOT CMAKE_BUILD_TYPE)` be added here to set a default CMAKE_BUILD_TYPE instead of the check added below?

Reply:
I think yes, setting default CMAKE_BUILD_TYPE to RelWithDebInfo when CMAKE_BUILD_TYPE is not set is better.